pkg/kubelet/dockertools/docker.go

```diff
@@ -180,11 +180,30 @@ func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 	}
 
 	if isDigested {
-		algo := digest.Digest().Algorithm().String()
-		sha := digest.Digest().Hex()
-		// Look specifically for short and long sha(s)
-		if strings.Contains(inspected.ID, algo+":"+sha) {
-			// We found the short or long SHA requested
+		for _, repoDigest := range inspected.RepoDigests {
+			named, err := dockerref.ParseNamed(repoDigest)
+			if err != nil {
+				glog.V(4).Infof("couldn't parse image RepoDigest reference %q: %v", repoDigest, err)
+				continue
+			}
+			if d, isDigested := named.(dockerref.Digested); isDigested {
+				// Allow prefix matches for shorter SHA values
+				if digest.Digest().Algorithm().String() == d.Digest().Algorithm().String() &&
+					strings.HasPrefix(digest.Digest().Hex(), d.Digest().Hex()) {
```

Comment:
I don't think `digest` or `d` will ever be a partial identifier. You can't get past `dockerref.ParseNamed` unless you have a full hash.